File: MusicApp/app/src/main/java/com/example/musicapp/MusicService.java
package com.example.musicapp;

import android.app.Service;
import android.content.Intent;
import android.media.MediaPlayer;
import android.os.IBinder;
import android.util.Log;

import androidx.annotation.Nullable;

public class MusicService extends Service {
    private final String TAG = MusicService.class.getSimpleName();
    private MediaPlayer mPlayer = null; // sample play for MediaPlayer

    @Nullable
    @Override
    public IBinder onBind(Intent intent) {
        // Return the communication channel to the service.
        // May return null if clients can not bind to the service.
        return null;
    }

    @Override
    public void onCreate() {
        super.onCreate();
        Log.d(TAG, "onCreate");

        // mp3 sample url
        // https://drive.google.com/file/d/1IDl3bUKMBOcpbvQu6I2g95iqnYyQ0qvf/view?usp=sharing
        mPlayer = MediaPlayer.create(this, R.raw.sample);   // play file: sample.mp3
        mPlayer.setLooping(true);
    }

    @Override
    public int onStartCommand(Intent intent, int flags, int startId) {
        Log.d(TAG, "onStartCommand");
        mPlayer.start(); // player start
        return super.onStartCommand(intent, flags, startId);
    }

    @Override
    public void onDestroy() {
        Log.d(TAG, "onDestroy");
        mPlayer.stop(); // player stop
        super.onDestroy();
    }
}

File: TestKotlin/app/src/main/java/com/tsi84/testkotlin/service/MusicService.kt
package com.tsi84.testkotlin.service

import android.app.Service
import android.content.Intent
import android.media.MediaPlayer
import android.os.IBinder
import android.util.Log
import com.tsi84.testkotlin.R

class MusicService: Service() {
    private val TAG = MusicService::class.java.simpleName
    private var mPlayer: MediaPlayer? = null

    override fun onBind(intent: Intent?): IBinder? {
        return null
    }

    override fun onCreate() {
        super.onCreate()
        Log.d(TAG, "onCreate")

        // mp3 sample url
        // https://file-examples-com.github.io/uploads/2017/11/file_example_MP3_700KB.mp3
        mPlayer = MediaPlayer.create(this, R.raw.sample)
        mPlayer!!.isLooping = true
    }

    override fun onDestroy() {
        Log.d(TAG, "onDestroy")
        if (mPlayer != null) {
            Log.d(TAG, "Music stopped..")
            mPlayer!!.stop()
        }
        super.onDestroy()
    }

    override fun onStartCommand(intent: Intent?, flags: Int, startId: Int): Int {
        if (mPlayer != null) {
            Log.d(TAG, "onStartCommand - Music start!!")
            mPlayer!!.start()
        }
        return super.onStartCommand(intent, flags, startId)
    }
}